Lf 2798 select the task location

Duncan-Brain opened the pull request:
[LF-2798](https://lite-farm.atlassian.net/browse/LF-2798?atlOrigin=eyJpIjoiNzczNzUzZWIwN2Y2NDBlYzgyYTRkMTkxODcyM2YxOGYiLCJwIjoiaiJ9) - 

My previous understanding was that crop enabled locations meant crop areas with current crops. The actual goal was to use any crop capable area regardless of current crops.

**Changes since previous merge:**
- Made a separate functional `<TaskIrrigationLocations />`  for the switch statement.
- Added the `readOnlyPinCoordinates()` so that wild crops are included as a location
- Handled the case where multi select is not enabled but also wild crops are selected
- Included title translations
- Refactored the goBack function to reuse for all

anant00729 commented on packages/webapp/src/components/Task/TaskLocations/index.jsx:
```diff
@@ -55,10 +55,13 @@ export default function PureTaskLocations({
-      LOCATIONS,
-      isMulti ? getSelectedLocations(location_id, selectedLocations) : [{ location_id }],
-    );
+    if(!isMulti && getValues('show_wild_crop')){
```
use early return 
```
const onSelectLocation = (location_id) => {
  if(!isMulti && getValues('show_wild_crop')) return;
  setValue(
    LOCATIONS,
    isMulti ? getSelectedLocations(location_id, selectedLocations) : [{ location_id }],
  );
};
```

Duncan-Brain replied:
@anant00729 Good catch! Can you check this again. 

I instead handled the !multi condition for the wild crop here. Now when a location is selected and the checkbox is also selected it will now deselect the checkbox and select the area to fit in with how the map multiselect works.

Duncan-Brain commented:
> @Duncan-Brain Please check the PR comments. Also, I would like to know the reason to remove isMulti as a prop from the component

So I added the isMulti prop to the component for my first PR and I reverted it back to what it was before.

anant00729 reviewed (approved):
This looks good to me. @wasswarichard Please review.

wasswarichard commented:
@Duncan-Brain it looks good.
